xServer.D/Feature/XDocuments/XDocumentClient.cs

```diff
@@ -0,0 +1,294 @@
+﻿using Common.Models.Graviex;
+using Common.Models.OrderBook;
+using Common.Utils;
+using Microsoft.Extensions.Logging;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Conventions;
+using MongoDB.Driver;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using x42.Feature.PowerDns;
+using x42.Feature.X42Client;
+using x42.Utilities;
+
+namespace x42.Feature.XDocuments
+{
+    public class XDocumentClient
+    {
+        private readonly MongoClient _client;
+        private readonly IMongoDatabase _db;
+        private readonly PowerDnsFeature _powerDnsService;
+        private X42Node _x42Client;
+        private X42ClientSettings _x42ClientSettings;
+
+        /// <summary>Instance logger.</summary>
+
+        /// <summary>
+        ///     A cancellation token source that can cancel the node monitoring processes and is linked to the <see cref="IxServerLifetime.ApplicationStopping"/>.
+        /// </summary>
+        private CancellationTokenSource _networkCancellationTokenSource;
+
+        /// <summary>Global application life cycle control - triggers when application shuts down.</summary>
+        private readonly IxServerLifetime _serverLifetime;
+
+        /// <summary>Factory for creating background async loop tasks.</summary>
+        private readonly IAsyncLoopFactory _asyncLoopFactory;
+        private readonly ILogger _logger;
+
+
+        public XDocumentClient(PowerDnsFeature powerDnsService, X42ClientSettings x42ClientSettings, ILoggerFactory loggerFactory, IxServerLifetime serverLifetime, IAsyncLoopFactory asyncLoopFactory)
+        {
+            var conventionPack = new ConventionPack { new CamelCaseElementNameConvention() };
+            ConventionRegistry.Register("camelCase", conventionPack, t => true);
+
+            _logger = loggerFactory.CreateLogger(GetType().FullName);
+
+            var mongoUser = Environment.GetEnvironmentVariable("MONGO_USER");
+            var mongoPassword = Environment.GetEnvironmentVariable("MONGO_PASSWORD");
+
+#if DEBUG
+            _client = new MongoClient($"mongodb://localhost:27017");
+#else
+            _client = new MongoClient($"mongodb://{mongoUser}:{mongoPassword}@xDocumentStore:27017/");
```

Comment:
Perhaps it would be better to allow the user to change the whole mongodb connection string.

Reply:
Yeah, will use just connection string going forward, need to refactor in a few places